src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -22,77 +45,1005 @@ const ActorInit Door_Warp1_InitVars = {
+    }
+    //! @bug SkelAnime_Free is not called for crystal variants
+}
+
+void DoorWarp1_SetupWarp(DoorWarp1* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->scale = 0;
+    this->unk_1AE = -140;
+    this->unk_1B0 = -80;
+    sWarpTimerTarget = 100;
+    this->unk_1BC = 1.0f;
+    this->lightRayAlpha = 0.0f;
+    this->warpAlpha = 0.0f;
+    this->crystalAlpha = 0.0f;
+
+    switch (this->actor.params) {
+        case WARP_YELLOW:
+        case WARP_ORANGE:
+        case WARP_GREEN:
+        case WARP_RED:
+            this->unk_194 = 0.23f;
+            this->unk_198 = 0.6f;
+            break;
+        case WARP_DESTINATION:
+            this->unk_194 = 0.0f;
+            this->unk_198 = 0.0f;
+            break;
+        case WARP_UNK_7:
+            this->scale = 100;
+            this->unk_1AE = 120;
+            this->unk_1B0 = 230;
+            this->unk_194 = 0.3f;
+            this->unk_198 = 0.3f;
+            break;
+        case WARP_BLUE_RUTO:
+        default:
+            this->unk_194 = 0.3f;
+            this->unk_198 = 0.3f;
+            break;
+    }
+
+    this->unk_19C = 0.0f;
+    this->actor.shape.yOffset = 1.0f;
+    this->warpTimer = 0;
+
+    switch (this->actor.params) {
+        case WARP_PURPLE_CRYSTAL:
+        case WARP_BLUE_RUTO:
+        case WARP_UNK_7:
+        default:
+            Lights_PointNoGlowSetInfo(&this->upperLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                      this->actor.world.pos.z, 200, 255, 255, 255);
+            Lights_PointNoGlowSetInfo(&this->lowerLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                      this->actor.world.pos.z, 200, 255, 255, 255);
+            break;
+        case WARP_CLEAR_FLAG:
+        case WARP_SAGES:
+        case WARP_YELLOW:
+        case WARP_DESTINATION:
+        case WARP_ORANGE:
+        case WARP_GREEN:
+        case WARP_RED:
+            break;
+    }
+
+    switch (this->actor.params) {
+        case WARP_CLEAR_FLAG:
+            DoorWarp1_SetupAction(this, DoorWarp1_AwaitClearFlag);
+            break;
+        case WARP_DESTINATION:
+            if ((!(gSaveContext.entranceIndex == 0x608 ||  // sacred forest meadow
+                   gSaveContext.entranceIndex == 0x564 ||  // death mountain crater
+                   gSaveContext.entranceIndex == 0x60C ||  // lake hylia
+                   gSaveContext.entranceIndex == 0x610 ||  // desert colossus
+                   gSaveContext.entranceIndex == 0x580) && // graveyard
+                 gSaveContext.sceneSetupIndex < 4) ||
+                (GET_PLAYER(globalCtx)->actor.params & 0xF00) != 0x200) {
+                Actor_Kill(&this->actor);
+            }
+            if (Actor_WorldDistXZToActor(&player->actor, &this->actor) > 100.0f) {
+                Actor_Kill(&this->actor);
+            }
+            DoorWarp1_SetupAction(this, DoorWarp1_WarpIn);
```

Comment:
```suggestion
            DoorWarp1_SetupAction(this, DoorWarp1_Destination);
```
I would keep the action `DoorWarp1_WarpIn` and the enum name `WARP_DESTINATION` consistent
"arrival" may be another candidate for describing this purpose of warp1 if you don't like "destination"?